Feat: polished whole website and added extra features

mdnm18 opened the pull request:
✅ UI/UX & Design Consistency

I established a more robust and consistent design system.

Fully Responsive Layout: Resolved all overflow issues on mobile and tablet devices for key components, including the Hero section, Countdown Timer, and Footer.

Reusable Button Component: Created a centralized <Button> component with support for variant, size, and font props. This ensures all buttons are consistent and easy to manage.

Component Refactoring: Refactored the Hero and Navbar sections to use the new reusable <Button> component, cleaning up the code and enforcing consistency.

Fluid & Consistent Typography:

Standardized all main section heading sizes.

Implemented fluid typography using CSS clamp() to ensure headings scale smoothly across all screen widths, not just at fixed breakpoints.

Enhanced Pop-up UI: Transformed the initial on-load pop-up from a code editor theme to a clean, glassmorphism-style call-to-action window for better user focus.

⚙️ Performance Optimization

The application's loading and rendering performance has been significantly improved.

Code Splitting: Implemented React.lazy() and <Suspense> to split the application into smaller JavaScript chunks. Components below the fold (like About Us, FAQs, Footer) are now loaded on demand, dramatically speeding up the initial page load time (Time to Interactive).

Image Lazy Loading: Added the loading="lazy" attribute to all images below the fold (Speaker and Sponsor sections), preventing them from being downloaded until the user scrolls near them.

Asset Optimization: The project is now prepared for its final pre-launch step: converting all .png images to the smaller, more efficient WebP format.

✨ Accessibility (A11y) & Animations

The site is now more accessible and provides a more polished user experience.

Semantic HTML: Improved the HTML structure in App.jsx and Timeline.jsx by using tags like <header> and <section>, making the site more understandable for screen readers and search engines.

Respect for User Motion Preferences: Implemented the useReducedMotion hook in the main animation component (FadeIn.jsx). This respects the user's system-level settings by disabling animations for those sensitive to motion.

Keyboard Navigation: Added highly visible focus-visible styles to all buttons, ensuring a clear and accessible experience for keyboard-only users.

Staggered Animations: Added staggered entrance animations to the FAQ list, making each item appear sequentially for a more polished and dynamic feel.

Enhanced Animation Flow: Implemented a re-animation trigger for the Hero section. The Hero animations now run for the first time after the initial pop-up is closed, providing a more impactful entrance to the main site.

darkdeathoriginal commented:
Overlapping issue in Event Timeline for tablet devices.

<img width="1535" height="2047" alt="localhost_5173_(iPad Mini)" src="https://github.com/user-attachments/assets/11fc4e6d-6eaa-4e85-a93b-30cbe6d2b896" />

darkdeathoriginal commented:
Inconsistent buttons for same action.
<img width="690" height="291" alt="image" src="https://github.com/user-attachments/assets/5c1f1fb1-bb73-4515-9eea-40ce209d3222" />

mdnm18 commented:
My primary contribution was to take the initial creative vision for the IGNISIA '25 website and systematically refactor and enhance it into a production-ready, industry-standard application. I focused on five key areas: Design System & UX, Performance Optimization, Accessibility, Code Quality, and Content Management.

**--- Refactoring for a Scalable & Consistent Design System**

I identified inconsistencies in the UI and refactored the codebase to be more maintainable and consistent.

1. What I Did: I created a single, reusable <Button> component from scratch.

Why I Did It: To eliminate code duplication, enforce design consistency, and centralize styling for all buttons on the site.

How I Did It: I built a highly versatile component accepting props for variant (primary/secondary), size (small/large), and font (pixel/nav), allowing it to adapt to different contexts like the Navbar and Hero sections.

2. What I Did: I implemented Fluid Typography for all major section headings.

Why I Did It: To move beyond fixed breakpoints and ensure headings scale smoothly across all screen sizes, always looking perfectly proportioned.

How I Did It: I replaced static responsive classes (e.g., text-4xl md:text-5xl) with a single, powerful CSS clamp() function via Tailwind's arbitrary value syntax.

**--- Enhancing User Experience & Interactivity**

I focused on elevating the user experience from a static page to a dynamic and interactive journey.

1. What I Did: I completely redesigned and re-engineered the Timeline component. (@aksharaagarwal21)

Why I Did It: The original design had significant layout bugs and lacked a "wow" factor. The goal was to create a unique, memorable centerpiece that was also perfectly responsive.

How I Did It: I rebuilt it as an interactive "roadmap" with a single-sided layout for mobile/tablet and an alternating desktop view. I fixed all alignment bugs and added refined animations, interactive nodes, and hover effects.

2. What I Did: I created and implemented a new on-load pop-up component. (@Punit173 )

Why I Did It: To immediately capture user attention and present a clear call-to-action upon visiting the site.

How I Did It: I built a responsive, glassmorphism-style modal that appears on page load and locks background scroll, and then integrated it into the main application state.

3. What I Did: I fixed all responsive layout issues for tablet (iPad) screens in the Navbar and Footer. (@darkdeathoriginal )

Why I Did It: The original layouts were breaking on intermediate screen sizes, causing content to overlap and become unusable.

How I Did It: I adjusted the Tailwind CSS grid breakpoints and component styles to create more appropriate two-column layouts for tablets, reserving the four-column designs for wider desktop screens.

**--- Implementing Critical Performance Optimizations**

I implemented several industry-standard techniques to make the website load faster and feel more fluid.

1. What I Did: I implemented Code Splitting.

Why I Did It: To drastically reduce the initial page load time.

How I Did It: I used React.lazy() and <Suspense> to ensure components below the fold (like the Footer and FAQ sections) are only downloaded when they are needed.

2. What I Did: I optimized the CountdownTimer component. (@darkdeathoriginal )

Why I Did It: To prevent unnecessary UI updates every second and improve the rendering efficiency.

How I Did It: I refactored the component by hoisting the TimeSegment and wrapping it in React.memo, ensuring that only the parts of the timer that change (like seconds) are re-rendered.

**---  Championing Accessibility & Modern Standards**

I ensured the website is accessible to all users and follows modern best practices.

1. What I Did: I improved Semantic HTML and Keyboard Navigation.

Why I Did It: To make the site fully usable for people who rely on screen readers or keyboard-only navigation.

How I Did It: I restructured the application using correct HTML tags (<header>, <section>) and added highly visible focus-visible styles to all interactive elements.

2. What I Did: I implemented support for the prefers-reduced-motion accessibility feature.

Why I Did It: To provide a safe and comfortable experience for users who are sensitive to motion.

How I Did It: I used Framer Motion's useReducedMotion hook to detect user system settings and automatically disable non-essential animations.

**--- Content Management & Code Quality**

I managed all content updates and improved the overall quality of the codebase.

1. What I Did: I updated the site's content to feature the two new guest speakers and new sponsors.

Why I Did It: To accurately reflect the final event details.

How I Did It: I refactored both the Speaker Section (creating a new two-speaker layout) and the Footer to dynamically handle multiple speakers. I also updated the Sponsors Section with new information, images, and styles.

2. What I Did: I finalized the "Register Now" functionality and cleaned the codebase.

Why I Did It: To complete the user journey and improve maintainability.

How I Did It: I linked all "Register Now" buttons to the official Unstop registration page. I also removed unnecessary files (PhoenixLogo.jsx, RisingEmbers.jsx) to clean up the project directory.

**I could use the DevOps tools**
